### PR TITLE
draft out tools for testing w/ mgohttp

### PR DESCRIPTION
**Pull request by natebrennand**
Trying to figure out how to best test "database" packages that utilize the `mgohttp` pkg...

### CODE REVIEW

File: mgohttptest/testing.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"context"
+	"sync"
+
+	"github.com/Clever/mgohttp/internal"
+	mgo "gopkg.in/mgo.v2"
+)
+
+// InjectSession returns a new session copy.
+func InjectSession(ctx context.Context, parentSession *mgo.Session, dbName string) context.Context {
```
Comment (natebrennand) [inline mgohttptest/testing.go]: option 1

File: mgohttptest/testing.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+// Config describes a mongo database that will be injected to the context
+type Config struct {
+	Name string
+	Sess *mgo.Session
+}
+
+// WithSessions calls a provided function with every provided database config injected to the
+func WithSessions(cfgs []Config, Do func(context.Context)) {
```
Comment (natebrennand) [inline mgohttptest/testing.go]: option 2

File: mgohttptest/testing.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func (t testHandler) Close() {
+	for _, s := range t.sessions {
+		s.Close()
+	}
+}
+
+// MakeContext creates a new Context that contains mgohttp database connections.
+func MakeContext(ctx context.Context, cfgs []Config) context.Context {
```
Comment (natebrennand) [inline mgohttptest/testing.go]: option 3

**Review by natebrennand — COMMENTED** [review]
@rgarcia, PTAL, figuring out how we can best test our code utilizing this

File: internal/contextkey.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Package internal exists so that we can share the Context key functions between
+// mgohttp & mgohttptest without making them public.
```
Comment (rgarcia) [inline internal/contextkey.go:2]: maybe include the reasons for why things are internal as comments

**Review by rgarcia — APPROVED** [review]

File: internal/contextkey.go (hunk 4 above)
Comment (natebrennand) [reply internal/contextkey.go:2]: Added in https://github.com/Clever/mgohttp/pull/2/commits/a956bb65ca8b0ad118ecdadceaf1422bb64e3043

**Pull request merged**